app/components/ai-agent.tsx

```diff
@@ -30,8 +47,301 @@ export const AIAgent = () => {
           }
         }
       });
+
+      // Append minimize button to the chat window
+      const chatDiv = document.querySelector("#n8n-chat");
```

Comment:
Could you refactor all of the document.querySelector code to use React state and CSS instead?

Reply:
I've updated the code using Refs, at least there's no redundant querySelectors inside the code.

Reply:
Ok, understood.